infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuInterClusterReplicationServer.java

```diff
@@ -305,9 +305,6 @@ private ServerContext getServerContext(Map<String, Object> opts ) {
         replicationDiscoveryService = new CorfuReplicationDiscoveryService(serverContext,
                 clusterManagerAdapter, discoveryServiceCallback);
 
-        Thread replicationDiscoveryThread = new Thread(replicationDiscoveryService, "discovery-service");
```

Comment:
I think it would be cleaner and easier to follow if we have a start method and the executor is started there, cause from here it is not obvious that the Discovery Service starts on the constructor...